apis/python/src/tiledbsc/soma_collection.py

```diff
@@ -210,33 +211,39 @@ def query(
         needn't specify these; if they don't, you must.
         """
 
-        soma_slices = []
-        for soma in self:
-            # E.g. querying for 'cell_type == "blood"' but this SOMA doesn't have a cell_type column in
-            # its obs at all.
-            if obs_query_string is not None and not soma.obs.has_attr_names(
-                obs_attrs or []
-            ):
-                continue
-            # E.g. querying for 'feature_name == "MT-CO3"' but this SOMA doesn't have a feature_name
-            # column in its var at all.
-            if var_query_string is not None and not soma.var.has_attr_names(
-                var_attrs or []
-            ):
-                continue
-
-            soma_slice = soma.query(
-                obs_attrs=obs_attrs,
-                var_attrs=var_attrs,
-                obs_query_string=obs_query_string,
-                var_query_string=var_query_string,
-                obs_ids=obs_ids,
-                var_ids=var_ids,
-            )
-            if soma_slice is not None:
-                # print("Slice SOMA from", soma.name, soma.X.data.shape(), "to", soma_slice.ann.X.shape)
-                soma_slices.append(soma_slice)
+        soma_slice_futures = []
+
+        with ThreadPoolExecutor() as executor:
```

Comment:
on a PR probably tomorrow I'll add the kwarg `max_workers=self._soma_options.max_workers` which will take just minutes to code up, and will be a reasonable boundedness and keep this from getting out of control for larger collections. perhaps defaulting `soma_options.max_workers` to 5 or 10.

Reply:
other than that this looks fine. i just pushed a commit with black/isort/flake8 mods to attempt CI pass.

Reply:
thanks @Shelnutt2 !